### ISSUE TITLE
proton: no examples or options for the AMD-specific backends, and pinning one under mode: cli captures an empty tree while the trial still exits 0

### ISSUE THREAD

**Issue by amd-vivekag** [ROCm/aorta#412]
The `proton` collector ships two examples (`examples/profiling/proton/triton-vecadd`, `examples/profiling/proton/triton-softmax`) and both leave `backend: auto`. Nothing in the repo shows what the individual AMD backends are *for*, and the option schema cannot express most of what upstream Proton offers for them.

The ask is three AMD-specific examples — one per AMD backend, each demonstrating that backend's distinct benefit — plus the option-schema work that makes naming an AMD backend usable at all.

"At all" is load-bearing. While scoping this I measured that pinning an explicit AMD backend under the default `mode: cli` captures **nothing**, deterministically, and the trial still exits 0 with no proton metrics. Backend examples are meaningless until that is guarded, so it is in scope here.

## How this was measured

- gfx950 (CDNA4), ROCm 7.0.2
- a ROCm venv with Triton 3.7.1 and PyTorch
- payload: the repo's own `examples/profiling/proton/triton-vecadd/vecadd.py`
- Proton invoked directly the same way `mode: cli` invokes it (`python -m triton.profiler.proton ...`), so the collector itself is not in the picture

## Finding A — an explicit AMD backend under `mode: cli` captures an empty tree

Three runs of the shipped payload, differing only in `-b`:

```bash
PAYLOAD=examples/profiling/proton/triton-vecadd/vecadd.py

# 1. no -b: what aorta's default `backend: auto` renders
python -m triton.profiler.proton -n /tmp/proton-auto \
    --context shadow --data tree "$PAYLOAD" --size 1048576 --iters 20

# 2. explicit roctracer
python -m triton.profiler.proton -n /tmp/proton-roctracer -b roctracer \
    --context shadow --data tree "$PAYLOAD" --size 1048576 --iters 20

# 3. explicit instrumentation
python -m triton.profiler.proton -n /tmp/proton-instr -b instrumentation \
    --context shadow --data tree "$PAYLOAD" --size 1048576 --iters 20
```

| `-b` | `.hatchet` size | tree contents |
|---|---|---|
| absent (`backend: auto`) | 3091 bytes | 27 kernels with real timings |
| `roctracer` | 160 bytes | `ROOT` only, `metrics` empty |
| `instrumentation` | 160 bytes | `ROOT` only, `metrics` empty |

Deterministic, 3/3 for each.

**Root cause** is in Proton's CLI, not in aorta. `_select_backend()` is called only when `-b` is *absent*, and that call initialises the Triton HIP driver as a side effect. `roctracer` records nothing unless its queue interceptor is installed after the HIP runtime is up:

```python
# triton/profiler/proton.py, Triton 3.7.1
def run_profiling(args, target_args):
    backend = args.backend if args.backend else _select_backend()

    start(args.name, context=args.context, data=args.data, backend=backend, hook=args.hook)
```

**The decisive experiment** isolates it to ordering alone, via the Python API on one payload:

- `proton.start(..., backend="roctracer")` first, `import torch` only afterwards → 160 bytes, empty tree.
- `triton.runtime.driver.active.get_current_target()` first — exactly what `_select_backend()` does — then `proton.start(..., backend="roctracer")` → 1649 bytes with real kernel data.

Nothing else differs between the two. Triton `main` has the same asymmetry for `roctracer` and `rocprofiler`.

**Consequence in aorta.** `parse_summary_from_streams` returns only `proton_artifact_dir` when the tree yields no timings (`src/aorta/instrumentation/proton/_parse.py:185`), so the trial exits 0 with no proton metrics at all — the silent-unprofiled outcome the collector's design rejects elsewhere. `docs/profiling-collectors.md:534` already carries a troubleshooting entry for a ROOT-only profile, but it frames the pin as a *hazard* ("pin one only when you need to"). The measurement says it is unconditional under `mode: cli`: the pin is not risky, it is a no-op.

## Finding B — shipped Triton drops `--mode`, so two existing options are silent no-ops

Triton 3.7.1's CLI parses `-m/--mode` and never forwards it to `start()` — visible in the snippet above, which has no `mode=`. Triton `main` does pass it.

So `instrumentation_mode` and `granularity` (`docs/profiling-collectors.md:217-218`) render a `--mode` argument that every released Triton discards. They are honoured only in `mode: env`, where the payload calls `proton.start()` itself. Today's docs claim neither limitation.

## Finding C — the schema cannot express the AMD backend options that do exist

Upstream Proton on `main` supports `pcsampling` / `periodic_flushing` as *backend* modes on `rocprofiler`, `periodic_flushing` on `roctracer`, and `-k/--hook triton`. `src/aorta/instrumentation/proton/_options.py` has no `hook` key, and its only `--mode` producer is gated to `backend: instrumentation` (`_options.py:128-134`), which is a different concept from a backend mode. A recipe cannot ask for any of them.

## Finding D — `rocprofiler` exists in no released Triton, and newer ROCm does not change that

`rocprofiler` is a Triton-version feature, not a ROCm one. Querying the images directly, every one offers `-b {cupti,roctracer,instrumentation}` and no `rocprofiler`:

| image | Triton |
|---|---|
| `rocm/pytorch:rocm7.14_ubuntu26.04_py3.14_pytorch_release_2.12.0` | 3.7.1 |
| `rocm/pytorch:latest` | 3.6.0+rocm7.2.4 |
| an AMD-internal ROCm 7.2 workload image | 3.7.0 |

PyPI's latest `triton` is 3.7.1; PyTorch's ROCm nightly tops out at `pytorch-triton-rocm` 3.6.0. On Triton `main`, `-b` choices come from `libproton.get_available_profilers()` and `rocprofiler` is present with modes `[None, "pcsampling", "periodic_flushing"]`. Reaching it means a source build. The `backend` row at `docs/profiling-collectors.md:214` is right that pinning is a version commitment; it just does not say the commitment is currently unsatisfiable for `rocprofiler`.

### Examples

Three directories under `examples/profiling/proton/`, each with the payload / `recipe.yaml` / `README.md` shape `tests/test_examples.py` already enforces, and each standalone-runnable, self-checking and cheap by default like the existing two.

All three use **`mode: env`, not `mode: cli`** — forced by Finding A, and what makes them verifiable today: an env-mode payload calls `proton.start()` itself after `import torch`, so the runtime is already initialised.

- **`amd-roctracer/`** — whole-kernel timing across a multi-kernel pipeline. Shows kernel-level attribution over a real launch sequence, and is the portable AMD path that works on every shipped Triton.
- **`amd-instrumentation/`** — an intra-kernel hotspot inside one deliberately unbalanced kernel, with `granularity: warp`. Shows attribution *inside* a kernel, and that this is the only backend that coexists with `rocprof` (no HSA queue interception).
- **`amd-rocprofiler/`** — the rocprofiler-sdk path with `backend_mode: pcsampling`. Given Finding D this one ships documented-but-unverified, with the Triton-`main` requirement stated plainly and the `get_available_profilers()` check to run first.

### Option schema

In `src/aorta/instrumentation/proton/_options.py`:

- `BACKEND_MODES: dict[str, frozenset[str]]` — `rocprofiler: {pcsampling, periodic_flushing}`, `roctracer: {periodic_flushing}`, `cupti: {pcsampling, periodic_flushing}`.
- New keys `backend_mode` (validated against `BACKEND_MODES[backend]`) and `hook` (`triton`).
- `instrumentation_mode` / `granularity` and their `backend: instrumentation` gate stay exactly as they are — no breakage, and the two concepts stay distinct.
- Reject `backend_mode` together with `instrumentation_mode`: they render into the same Proton `--mode`.

In `src/aorta/instrumentation/proton/__init__.py`: `mode_argument` renders `backend_mode`, `build_argv_prefix` emits `-k <hook>`, `build_env` exports `AORTA_PROTON_HOOK`.

### The Finding A guard

`wrap_argv` raises `ProtonWrapError` for `mode: cli` with an explicit `backend` of `roctracer` or `rocprofiler`, stating the reason (Proton's CLI initialises the runtime only when `-b` is omitted, so the capture would come back empty) and naming `mode: env` as the route. `backend: auto` is untouched — it is the working path — and so is `instrumentation`, which is not queue-intercepting.

An empty capture that exits 0 is worse than a load-time error, which is the same reasoning that already makes `instrumentation_mode` on the wrong backend an error rather than accepted-and-ignored.

### Docs

- `docs/profiling-collectors.md`: `backend_mode` / `hook` rows; a "pinning an explicit AMD backend" subsection carrying Finding A and the `mode: env` requirement; correct the `backend` row (line 214) and sharpen the ROOT-only troubleshooting entry (line 534) from hazard to unconditional; note Finding B on the `instrumentation_mode` / `granularity` rows.
- `examples/profiling/README.md`: three new table rows, and extend the backend paragraph.
- Both existing example READMEs understate pinning the same way (`proton/triton-vecadd/README.md:94-101`, `proton/triton-softmax/README.md:96-98`); fix both.

### Tests

- `tests/instrumentation/test_proton.py`: `backend_mode` accept/reject per backend, `hook` rendering in both attach modes, the `backend_mode` + `instrumentation_mode` conflict, and the new guard (raises for explicit `roctracer`/`rocprofiler`, silent for `auto` and `instrumentation`).
- `tests/instrumentation/test_proton_smoke_gpu.py`: a GPU smoke test asserting an env-mode explicit-`roctracer` capture yields a **non-empty** tree. Asserting a `.hatchet` merely exists is exactly the trap Finding A falls into — that is the regression which would have caught it.

## Out of scope: two upstream Triton bugs

Findings A and B are defects in Triton's Proton CLI, not in aorta:

- `run_profiling` calls `_select_backend()` only when `-b` is absent, so an explicit backend loses the runtime initialisation it depends on and captures nothing.
- `-m/--mode` is parsed and never forwarded to `start()`.

Both are worth reporting upstream separately. This issue only works around A (guard plus `mode: env`) and documents B; neither upstream fix is a dependency for the work above.
